### ISSUE TITLE
Missing to_matrix_cl for std::vector<T>

### ISSUE THREAD

**Issue by rok-cesnovar** [stan-dev/math#1400]
## Description

I discovered we are missing `to_matrix_cl()` for ~~a scalar and~~ std::vector ~~and both of~~ which we need for some of the GPU GLMs to work with the way this will run with the new compiler. 

I would consider this a bugfix so hopefully it does not get locked down with the feature freeze. I will fix it immediately but not sure when it will get review/pass Jenkins.

## Example
The following should work:
```
std::vector<double> d1_cpu{ 10.0, 20.0, 30.0 }; 
stan::math::matrix_cl<double> d11_cl = to_matrix_cl(d1_cpu);
```

#### Current Version:
v2.20.0

**Comment by rok-cesnovar** [comment]
I am going to solve this as part of #1356 because this will need the temporaries bugfix.

**Comment by rok-cesnovar** [comment]
Fixed via #1356
